crates/cli/src/bin/wasm-bindgen-test-runner/node.rs

```diff
@@ -88,7 +98,14 @@ pub fn execute(
     ",
     );
 
-    let js_path = tmpdir.join("run.js");
+    let js_path = if module_format {
+        // fixme: this is a hack to make node understand modules
+        let package_json = tmpdir.join("package.json");
+        fs::write(&package_json, r#"{"type": "module"}"#).unwrap();
+        tmpdir.join("run.mjs")
```

Comment:
> It looks like for use-cases without a `package.json` the explicit extension is really useful 😄
> 
> Test is running now, I added a `package.json` temporarily to make the test work. It can be removed with the extension change.

So coming back to the discussion from earlier in https://github.com/rustwasm/wasm-bindgen/pull/4027#issuecomment-2263042667:
> > Is there a reason file extensions should be consistent across targets?
> 
> Principle of least surprise is the main one, but I don't have reasons beyond that.

What I was trying to say is: consistency only makes sense when the treatment of the output is desired to be similar. But it seems to me that you won't use the modules target *and* the no-modules target, you will be using either.

Consistency would be more desirable if, lets say, you want to use *both* targets with the same configuration.

Please keep in mind that I barely touched this ecosystem in my life and I'm clueless about the tools. I might have some assumptions that are just wrong.

---

I'm in favor of switching the target right now if we can avoid the `package.json`.
You mentioned some issues before with switching to the new file, are these resolved at this point seeing that the test runs successfully?

Reply:
I just didn't consider the use-case of running bare scripts without `package.json` as very common, the vast majority of scripts I've seen are distributed as packages, but of course it's a use-case that exists. In a package it just doesn't matter as long as all files are in the same format. But I don't disagree.

> I'm in favor of switching the target right now if we can avoid the package.json. [...]

There is still a bunch of code that assumes a `.js` extension, I haven't checked how much. Right now only the `run.js` is renamed, since it's not used anywhere else.

Reply:
> There is still a bunch of code that assumes a `.js` extension, I haven't checked how much. Right now only the `run.js` is renamed, since it's not used anywhere else.
> 
> I think the `package.json` is fine here for now since it's only in a temporary directory and only in the test runner, and it's only created when running tests for the new target. I can get back to it during the week.

No worries, lets do it in a follow-up then!